manager/controllers/app/m4dapplication_controller.go

```diff
@@ -232,6 +232,35 @@ func (r *M4DApplicationReconciler) deleteExternalResources(applicationContext *a
 	return nil
 }
 
+// getReadModulesEndpoints populates the ReadEndpointsMap map in the status of the m4dapplication
+// Current implementation assumes there is only one cluster with read modules (which is the same cluster the user's workload)
+func (r *M4DApplicationReconciler) getReadModulesEndpoints(applicationContext *app.M4DApplication, blueprintsMap map[string]app.BlueprintSpec, moduleMap map[string]*app.M4DModule) {
```

Comment:
Minor thing. The function name is misleading. It does not get the read module endpoints. It sets them. Also there is no need to bind this to `M4DApplicationReconciler`.
This could be e.g. `setReadModuleEndpoints`. And then the call below would be:
```
setReadModuleEndpoints(applicationContext, blueprintPerClusterMap, moduleMap)
```

Reply:
Fixed